app/controllers/no_businesses_controller.rb

```diff
@@ -4,8 +4,13 @@ class NoBusinessesController < ApplicationController
   def new; end
 
   def create
-    submission = @task.no_business.first
-    redirect_to task_submission_path(task_id: @task.id, id: submission.id)
+    if params[:replacement]
```

Comment:
When I tried that, the request for the non-correction submission ends up having the shape

```
stub_request(:post, "http://localhost:3000/v1/tasks/2d98639e-5260-411f-a5ee-61847a2e067c/no_business").
         with(
           body: "{\"correction\":false}",
           headers: {
       	  'Accept'=>'application/vnd.api+json',
       	  'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
       	  'Content-Type'=>'application/vnd.api+json',
       	  'User-Agent'=>'Faraday v0.12.2',
       	  'X-Auth-Id'=>'auth0|123456'
           }).
         to_return(status: 200, body: "", headers: {})
```
JSONAPI adds ` body: "{\"correction\":false}"` forever and ever to the "regular" path, and that seems weird and suboptimal.

Reply:
Personally I do not have a problem with the stub for a regular no-business correction being updated to include `body: "{\"correction\":false}"`. That is a valid API request, and the API will respond as we'd expect. But I won't push on the point, it's not a major blocker, I just think it would make for slightly neater controller action code..